background.js

```diff
@@ -168,6 +159,45 @@ function notify_state_change() {
     });
 }
 
+function decorate_avatar(imgdata, color) {
+    if (color === undefined) return imgdata;
+
+    let buffer = new OffscreenCanvas(48, 48);
+    let ctx_buffer = buffer.getContext("2d");
+    ctx_buffer.putImageData(imgdata, 0, 0);
+
+    let canvas = new OffscreenCanvas(48, 48);
+    let ctx = canvas.getContext("2d");
+    ctx.save();
+    ctx.beginPath();
+    ctx.arc(24, 24, 18, 0, Math.PI * 2, false);
+    ctx.clip();
+    ctx.drawImage(buffer, 0, 0);
+    ctx.restore();
+    ctx.save();
+    ctx.strokeStyle = color;
+    ctx.lineWidth = 3;
+    ctx.beginPath();
+    ctx.arc(24, 24, 22, 0, Math.PI * 2, false);
+    ctx.stroke();
+    ctx.restore();
+    return ctx.getImageData(0, 0, 48, 48);
```

Comment:
Can we do the circle slightly thinker? Depending on your avatar pic, the difference can be hard to spot.

Reply:
It looks like this is an artifact of multiple fractional scalings (+anti aliasing). This especially becomes problematic on 2.5k or 4k screens. Therefor you can provide multiple versions of an icon, based on size. I'll change the code to generate these versions directly, instead of downsampling a 48x48px version.

Reply:
@jan-kiszka That was surprisingly hard to fix, but now we avoid the repeated downscaling of the icon, leading to a much sharper image in general. While fixing this, I also noticed that we call out to the broker too often (this behavior worsened with the tracking of the online / offline state). This is fixed in another commit.